### DISCUSSION TITLE
Visual exploration of event streams

### DISCUSSION THREAD

**Discussion by jakubpnf** [MaterializeInc/mz-hack-day-2022#11] [Show and tell]
Together with my friend @marusek we developed a POC of integrating `Materialize`+`dbt`+`Redpanda` with our data exploration tool we've been experimenting for some time - `nPond`.

![Screen Recording 2022-02-18 at 12 34 42](https://user-images.githubusercontent.com/98587630/154682899-7a406f1d-2f52-41a6-9987-51a32df05077.gif)

To visualize data with `nPond`:
1.  We translated a DAG that can be consumed by `nPond` into a `Materialize` query (view)
2. The query enriches events with metadata needed for visualisation 
3. Enriched data is written to a `Redpada` sink. 
4. `nPond` reads from the topic and draws events
![materialize+npond](https://user-images.githubusercontent.com/98587630/154682866-39a1928f-fc02-4faf-ba2d-7196fb6e634c.jpg)

The query and sink: https://github.com/jakubpnf/mz-hack-day-2022/commit/ff94823b8127ff044e99dc67bc5ae79c594ef52b
The DAG for `nPond`:
```
{
  id: 0,
  childNodes: [
    {
      id: 1,
      label: 'Air',
      childNodes: [
        {
          id: 3,
          label: 'US',
          childNodes: null,
        },
        {
          id: 5,
          label: 'China',
          childNodes: null,
        },
        {
          id: 4,
          label: 'Rest of the world',
          childNodes: null,
        },
      ],
    },
    {
      id: 2,
      label: 'Ground',
      childNodes: null,
    },
  ],
}
```

**Comment by christineberger** [comment]
Wow, this is really cool @jakubpnf! 🪐
